docs/source/text_processing/g2p/g2p.rst

```diff
@@ -0,0 +1,203 @@
+
+
+
+.. _bert_heteronym_cl:
+
+Purpose-built BERT-based classification model for heteronym disambiguation
+--------------------------------------------------------------------------
+
+HeteronymClassificationModel is a BERT-based :cite:`g2p--ddevlin2018bert` model represents a token classification model and can handle multiple heteronyms at once. The model takes a sentence as an input, and then for every word, it selects a heteronym option out of the available forms.
+We mask irrelevant forms to disregard the model’s predictions for non-ambiguous words. E.g., given  the input “The Poems are simple to read and easy to comprehend.” the model scores possible {READ_PRESENT and READ_PAST} options for the word “read”.
+Possible heteronym forms are extracted from the WikipediaHomographData :cite:`g2p--gorman2018improving`.
+
+The model expects input to be in `.json` manifest format, where is line contains at least the following fields:
+
+.. code::
+
+  {"text_graphemes": "Oxygen is less able to diffuse into the blood, leading to hypoxia.", "start_end": [23, 30], "homograph_span": "diffuse", "word_id": "diffuse_vrb"}
+
+Manifest fields:
+
+* `text_graphemes` - input sentence
+
+* `start_end` - beginning and end of the heteronym span in the input sentence
+
+* `homograph_span` - heteronym word in the sentence
+
+* `word_id` - heteronym label
+
+To convert the WikipediaHomographData to `.json` format suitable for the HeteronymClassificationModel training, run:
+
+.. code-block::
+
+    # WikipediaHomographData could be downloaded from `https://github.com/google-research-datasets/WikipediaHomographData <https://github.com/google-research-datasets/WikipediaHomographData>`__.
+
+    python NeMo/scripts/dataset_processing/g2p/export_wikihomograph_data_to_manifest.py \
+            --data_folder=<Path to WikipediaHomographData>/WikipediaHomographData-master/data/eval/
+            --output=eval.json
+    python NeMo/scripts/dataset_processing/g2p/export_wikihomograph_data_to_manifest.py \
+            --data_folder=<Path to WikipediaHomographData>/WikipediaHomographData-master/data/train/
+            --output=train.json
+
+To train and evaluate the model, run:
+
+.. code-block::
+
+    python heteronym_classification_train_and_evaluate.py \
+        train_manifest=<Path to manifest file>" \
+        validation_manifest=<Path to manifest file>" \
+        model.encoder.pretrained="<Path to .nemo file or pretrained model name from list_available_models()>" \
+        do_training=True \
+        do_testing=True
+
+
+To run inference with a pretrained HeteronymClassificationModel, run:
+
+.. code-block::
+
+    python heteronym_classification_inference.py \
+        manifest="<Path to .json manifest>" \
+        pretrained_model="<Path to .nemo file or pretrained model name from list_available_models()>" \
+        output_file="<Path to .json manifest to save prediction>"
```

Comment:
Similarly here--what does the output look like/what should the user look for?

Reply:
added